File: task/src/main/java/com/goexp/galgame/data/db/importor/mysql/GameDB.java
package com.goexp.galgame.data.db.importor.mysql;

import com.goexp.common.db.mysql.DBOperatorTemplate;
import com.goexp.galgame.data.model.Game;

import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.List;

public class GameDB extends DBOperatorTemplate {


    public void insert(Game game) {
        try {

            runner.update("insert into game_copy1 (id,`name`,brandid,publishdate,smallimg)values (?,?,?,?,?)",
                    game.id
                    , game.name
                    , game.brandId
                    , game.publishDate.toString()
                    , game.smallImg
            );


        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

    public void update(Game game) {
        try {

            runner.update("update game_copy1 set `publishDate`=?,smallimg=? where id=?"
                    , game.publishDate.toString()
                    , game.smallImg
                    , game.id
            );


        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

//    public void updateDate(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `publishDate`=? where id=?"
//                    , game.publishDate.toString()
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//
//    public void updateTag(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `tag`=? where id=?"
//                    , game.tag
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//
//    public void batchUpdateAll(List<Game> games) {
//        try {
//
//            var param = new Object[games.size()][];
//
//            for (var i = 0; i < games.size(); i++) {
//                var game = games.get(i);
//                param[i] = new Object[]{game.painter, game.writer, game.type, game.tag, game.story, game.id};
//            }
//
//            runner.batch("update game_copy1 set `painter`=?,writer=?,`type`=?,tag=?,story=? where id=?", param);
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }

    public void updateAll(Game game) {
        try {

            runner.update("update game_copy1 set `painter`=?,writer=?,`type`=?,tag=?,story=?,intro=?,brandId=? where id=?"
                    , game.painter
                    , game.writer
                    , game.type
                    , game.tag
                    , game.story
                    , game.intro
                    , game.brandId
                    , game.id
            );


        } catch (SQLException e) {
            e.printStackTrace();
        }
    }

//    public void updateBrand(Game game) {
//        try {
//
//            runner.update("update game_copy1 set brandId=? where id=?"
//                    , game.brandId
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//
//    public void updateSmallImg(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `smallimg`=? where id=?"
//                    , game.imgUrl
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//    public void updateTitle(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `name`=? where id=?"
//                    , game.name
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//    public void updateStory(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `story`=? where id=?"
//                    , game.story
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//
//    public void updateIntro(Game game) {
//        try {
//
//            runner.update("update game_copy1 set `intro`=? where id=?"
//                    , game.intro
//                    , game.id
//            );
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }
//
//    public void updateStory(List<Game> games) {
//        try {
//
//            var param = new Object[games.size()][];
//
//            for (var i = 0; i < games.size(); i++) {
//                var g = games.get(i);
//                param[i] = new Object[]{g.story, g.id};
//            }
//
//            runner.batch("update game_copy1 set `story`=? where id=?", param);
//
//
//        } catch (SQLException e) {
//            e.printStackTrace();
//        }
//    }


    public boolean exist(int id) {
        try {

            return runner.query("select 1 from game_copy1 where id=? limit 1", ResultSet::next, id);


        } catch (SQLException e) {
            e.printStackTrace();
        }

        return false;
    }


    public static class Star {
        public void insert(Game game) {
            try {
                runner.update("insert into game_star (id) values (?)", game.id);
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

    }

    public static class State {
        public void insert(Game game) {
            try {
                runner.update("insert into game_state (id) values (?)", game.id);
            } catch (SQLException e) {
                e.printStackTrace();
            }
        }

    }

}

File: task/src/main/scala/com/goexp/galgame/data/source/getchu/importor/GameDB.scala
package com.goexp.galgame.data.source.getchu.importor

import com.goexp.common.db.mongo.DBOperatorTemplate
import com.goexp.galgame.common.model.game.GameState
import com.goexp.galgame.data.model.{Brand, Game}
import com.goexp.galgame.data.source.getchu.DB_NAME
import com.goexp.galgame.data.source.getchu.query.GameQuery
import com.mongodb.client.model.Filters
import com.mongodb.client.model.Filters.and
import com.mongodb.client.model.Updates.{combine, set}
import org.bson.Document

import scala.jdk.CollectionConverters._

object GameDB {
  val tlp = new DBOperatorTemplate(DB_NAME, "game")

  def insert(game: Game) = {
    val gameDoc = new Document("_id", game.id)
      .append("name", game.name)
      .append("publishDate", game.publishDate)
      .append("smallImg", game.smallImg)
      .append("state", 0)
      .append("star", 0)
      .append("state", game.state.value)
      .append("brandId", game.brandId)
      .append("isNew", game.isNew)

    tlp.exec(documentMongoCollection => {
      documentMongoCollection.insertOne(gameDoc)
    })
  }

  def update(game: Game): Unit =
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(
        Filters.eq(game.id),
        combine(
          set("publishDate", game.publishDate),
          set("smallImg", game.smallImg)
        )
      )
    })

  def update(id: Int, middle: Int, website: String, group: String): Unit =
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(
        Filters.eq(id),
        combine(
          set("middle", middle),
          set("website", website),
          set("group", group)
        )
      )
    })

  def updateAll(game: Game) =
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(
        Filters.eq(game.id),
        combine(
          set("painter", game.painter),
          set("writer", game.writer),
          set("type", game.`type`),
          set("tag", game.tag),
          set("story", game.story),
          set("intro", game.intro),
          set("brandId", game.brandId))
      )
    })

  def updateChar(game: Game) = {
    val gameCharDocs = game.gameCharacters
      .asScala.to(LazyList)
      .map(person => {
        new Document("name", person.name)
          .append("intro", person.intro)
          .append("cv", person.cv)
          .append("truecv", person.trueCV)
          .append("img", person.img)
          .append("index", person.index)
      }).asJava

    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(Filters.eq(game.id), set("gamechar", gameCharDocs))
    })
  }

  def updateImg(game: Game) = {
    val imgdocs = game.gameImgs
      .asScala.to(LazyList)
      .map(img => {
        new Document("src", img.src)
          .append("index", img.index)
      }).asJava
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateOne(Filters.eq(game.id), set("simpleImg", imgdocs))
    })
  }

  def blockAllGame(item: Brand) =
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateMany(
        and(
          Filters.eq("brandId", item.id),
          Filters.ne("state", GameState.SAME.value),
          Filters.ne("state", GameState.BLOCK.value)
        )
        , set("state", GameState.BLOCK.value))

    })

  def resetState(item: Brand) =
    tlp.exec(documentMongoCollection => {
      documentMongoCollection.updateMany(
        Filters.eq("brandId", item.id)
        , set("state", GameState.UNCHECKED.value))

    })

  def exist(id: Int): Boolean = GameQuery.fullTlp.where(Filters.eq(id)).exists


  object StateDB {
    def update(game: Game) =
      tlp.exec(documentMongoCollection => {
        documentMongoCollection.updateOne(Filters.eq(game.id), set("state", game.state.value))
      })
  }

}

